cadc-vos-server-alt/src/main/java/org/opencadc/vospace/server/NodeFault.java

```diff
@@ -83,51 +86,62 @@
  */
 public enum NodeFault {
     // IVOA Standard Faults - not an exhaustive list
-    PermissionDenied(new NotAuthenticatedException(VOS.IVOA_FAULT_PERMISSION_DENIED)),
-    InvalidURI(new IllegalArgumentException(VOS.IVOA_FAULT_INVALID_URI)),
-    NodeNotFound(new ResourceNotFoundException(VOS.IVOA_FAULT_NODE_NOT_FOUND)),
-    DuplicateNode(new ResourceAlreadyExistsException(VOS.IVOA_FAULT_DUPLICATE_NODE)),
-    InvalidToken(new IllegalArgumentException(VOS.IVOA_FAULT_INVALID_TOKEN)),
-    InvalidArgument(new IllegalArgumentException(VOS.IVOA_FAULT_INVALID_ARG)),
-    TypeNotSupported(new IllegalArgumentException(VOS.IVOA_FAULT_TYPE_NOT_SUPPORTED)),
-    // Other Faults
-    ContainerNotFound(new ResourceNotFoundException(VOS.CADC_FAULT_CONTAINER_NOT_FOUND)),
-    //TODO not sure still needed.
-    //    RequestEntityTooLarge(new ByteLimitExceededException(VOS.CADC_FAULT_REQUEST_TOO_LARGE)),
-    UnreadableLinkTarget(new ResourceNotFoundException(VOS.CADC_FAULT_UNREADABLE_LINK)),
-    ServiceBusy(new TransientException(VOS.CADC_FAULT_SERVICE_BUSY)),
-    NodeLocked(new ResourceLockedException(VOS.CADC_FAULT_NODE_LOCKED)),
-    NotAuthenticated(new NotAuthenticatedException("NotAuthenticated"));
+    PermissionDenied(VOS.IVOA_FAULT_PERMISSION_DENIED),
+    InvalidURI(VOS.IVOA_FAULT_INVALID_URI),
+    NodeNotFound(VOS.IVOA_FAULT_NODE_NOT_FOUND),
+    DuplicateNode(VOS.IVOA_FAULT_DUPLICATE_NODE),
+    InvalidToken(VOS.IVOA_FAULT_INVALID_TOKEN),
+    InvalidArgument(VOS.IVOA_FAULT_INVALID_ARG),
+    TypeNotSupported(VOS.IVOA_FAULT_TYPE_NOT_SUPPORTED),
+    // CADC defined faults
+    ContainerNotFound(VOS.CADC_FAULT_CONTAINER_NOT_FOUND),
+    RequestEntityTooLarge(VOS.CADC_FAULT_REQUEST_TOO_LARGE),
+    UnreadableLinkTarget(VOS.CADC_FAULT_UNREADABLE_LINK),
+    ServiceBusy(VOS.CADC_FAULT_SERVICE_BUSY),
+    NodeLocked(VOS.CADC_FAULT_NODE_LOCKED),
+    NotAuthenticated(VOS.CADC_FAULT_NOT_AUTHENTICATED);
 
-    private Exception status;
-    private String message;
+    private String status;
     private boolean serviceFailure;
 
-    private NodeFault(Exception ex) {
-        this.status = ex;
+    private NodeFault(String fault) {
+        this.status = fault;
         this.serviceFailure = false;
     }
 
-    public Exception getStatus() {
-        return status;
-    }
-
-    public String toString() {
-        return name();
-    }
-
-    public String getMessage() {
-        if (message != null) {
-            return message;
+    public Exception getStatus(String msg) {
+        String exMsg = status;
+        if (StringUtil.hasText(msg)) {
+            exMsg += ": " + msg;
         }
-        if (status != null) {
-            return status.getMessage();
+        switch (status) {
+            case VOS.IVOA_FAULT_PERMISSION_DENIED:
```

Comment:
permission denied should be AccessControlException

there is standard exception -> http status code mapping in cadc-rest and standard status code -> exception mapping in cadc-util  (HttpTransfer.checkErrors) that is supposed to reverse it